src/plugins/data_views/server/rest_api_routes/public/swap_references.ts

```diff
@@ -27,8 +27,10 @@ interface GetDataViewArgs {
 
 interface SwapRefResponse {
   result: Array<{ id: string; type: string }>;
-  preview: boolean;
-  deleteSuccess?: boolean;
+  deleteStatus?: {
```

Comment:
Do we have a reliable way to test this functionality? It would be great to have a test that also covers what happens if the delete fails/has refs if possible.

Reply:
I added a test. I realized that there's a simpler valid use case for this - when you swap some refs, but not all, and ask it to delete.

Reply:
Ah that makes sense and covers exactly what I was looking for. Thanks!